www/app/locals/local/service.js

```diff
@@ -0,0 +1,60 @@
+(function() {
+  'use strict';
+
+  angular
+  .module('musicHack.locals')
+  .service('VenuePlayerService', VenuePlayerService);
+
+  VenuePlayerService.$inject = ['$rootScope', 'env', '$firebaseArray'];
+
+  /* @ngInject */
+  function VenuePlayerService($rootScope, env, $firebaseArray) {
+    this.fetchQueue = fetchQueue;
+    this.onYouTubeIframeAPIReady = onYouTubeIframeAPIReady;
+
+    var ref = new Firebase(env.firebaseApiUrl);
+    var uid = $rootScope.authenticatedUser.uid;
+    var upcomingSongs = [
```

Comment:
This should be handled on the controller with the service's help, the same for the ended event logic.